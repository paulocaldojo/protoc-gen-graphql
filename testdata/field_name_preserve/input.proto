syntax = "proto3";

package protoc_gen_graphql.test.basic;

option go_package = "github.com/paulocaldojo/protoc-gen-graphql/testdata/field_name_preserve";

import "graphql/options.proto";

service FooService {
  option (graphql.service) = { reference_name: "foo_service" };

  rpc BarMethod(ScalarFields) returns (ScalarFields) {
    option (graphql.method) = { operation: "query" };
  }

  rpc BazMethod(ScalarFields) returns (ScalarFields) {
    option (graphql.method) = { operation: "query" };
  }
}

message ScalarFields {
  string string_field = 1;
  bytes bytes_field = 2;
  int32 int32_field = 3;
  uint32 uint32_field = 4;
  sint32 sint32_field = 5;
  fixed32 fixed32_field = 6;
  int64 int64_field = 7;
  uint64 uint64_field = 8;
  sint64 sint64_field = 9;
  fixed64 fixed64_field = 10;
  float float_field = 11;
  double double_field = 12;
  bool bool_field = 13;
}

enum EnumType {
  UNKNOWN = 0;
  FIRST = 1;
  SECOND = 2;
}

enum UnusedEnum {
  ZERO = 0;
}

message ComplexFields {
  ScalarFields message_field = 1;
  EnumType enum_field = 2;
  repeated string repeated_scalar_field = 3;
  repeated ScalarFields repeated_message_field = 4;
  map<string, EnumType> map_field = 5;
  oneof oneof_field {
    string first = 6;
    ScalarFields second = 7;
  }
  Empty empty_field = 8;
}

message Outer {
  message Middle {
    message Inner {
      string inner_name = 1;
    }
    string middle_name = 1;
  }
  Middle middle = 1;
  Middle.Inner inner = 2;
}

message Empty {}
