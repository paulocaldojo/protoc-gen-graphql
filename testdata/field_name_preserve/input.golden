# DO NOT EDIT! Generated by protoc-gen-graphql.

extend type Query {
  foo_service: ProtocGenGraphqlTestBasic_FooService_Query!
}

type ProtocGenGraphqlTestBasic_FooService_Query {
  BarMethod(input: ProtocGenGraphqlTestBasic_ScalarFieldsInput!): ProtocGenGraphqlTestBasic_ScalarFields
  BazMethod(input: ProtocGenGraphqlTestBasic_ScalarFieldsInput!): ProtocGenGraphqlTestBasic_ScalarFields
}

type ProtocGenGraphqlTestBasic_ScalarFields {
  string_field: String!
  bytes_field: String!
  int32_field: Int!
  uint32_field: Int!
  sint32_field: Int!
  fixed32_field: Int!
  int64_field: Int!
  uint64_field: Int!
  sint64_field: Int!
  fixed64_field: Int!
  float_field: Float!
  double_field: Float!
  bool_field: Boolean!
}

input ProtocGenGraphqlTestBasic_ScalarFieldsInput {
  string_field: String
  bytes_field: String
  int32_field: Int
  uint32_field: Int
  sint32_field: Int
  fixed32_field: Int
  int64_field: Int
  uint64_field: Int
  sint64_field: Int
  fixed64_field: Int
  float_field: Float
  double_field: Float
  bool_field: Boolean
}

type ProtocGenGraphqlTestBasic_ComplexFields {
  message_field: ProtocGenGraphqlTestBasic_ScalarFields
  enum_field: ProtocGenGraphqlTestBasic_EnumType!
  repeated_scalar_field: [String!]!
  repeated_message_field: [ProtocGenGraphqlTestBasic_ScalarFields!]!
  map_field: [ProtocGenGraphqlTestBasic_ComplexFields_MapFieldEntry!]!
  oneof_field: ProtocGenGraphqlTestBasic_ComplexFields_OneofFieldOneof
  empty_field: ProtocGenGraphqlTestBasic_Empty
}

"""
`ProtocGenGraphqlTestBasic_ComplexFields_OneofFieldOneof` represents the `oneof_field` oneof in `protoc_gen_graphql.test.basic.ComplexFields`.
"""
union ProtocGenGraphqlTestBasic_ComplexFields_OneofFieldOneof = ProtocGenGraphqlTestBasic_ComplexFields_OneofFieldOneof_First | ProtocGenGraphqlTestBasic_ComplexFields_OneofFieldOneof_Second

"""
`ProtocGenGraphqlTestBasic_ComplexFields_OneofFieldOneof_First` represents the `first` oneof field in `protoc_gen_graphql.test.basic.ComplexFields`.
"""
type ProtocGenGraphqlTestBasic_ComplexFields_OneofFieldOneof_First {
  _typename: String
  first: String!
}

"""
`ProtocGenGraphqlTestBasic_ComplexFields_OneofFieldOneof_Second` represents the `second` oneof field in `protoc_gen_graphql.test.basic.ComplexFields`.
"""
type ProtocGenGraphqlTestBasic_ComplexFields_OneofFieldOneof_Second {
  _typename: String
  second: ProtocGenGraphqlTestBasic_ScalarFields
}

"""
`ProtocGenGraphqlTestBasic_ComplexFields_MapFieldEntry` represents the `map_field` map in `protoc_gen_graphql.test.basic.ComplexFields`.
"""
type ProtocGenGraphqlTestBasic_ComplexFields_MapFieldEntry {
  key: String!
  value: ProtocGenGraphqlTestBasic_EnumType!
}

type ProtocGenGraphqlTestBasic_Outer {
  middle: ProtocGenGraphqlTestBasic_Outer_Middle
  inner: ProtocGenGraphqlTestBasic_Outer_Middle_Inner
}

type ProtocGenGraphqlTestBasic_Outer_Middle {
  middle_name: String!
}

type ProtocGenGraphqlTestBasic_Outer_Middle_Inner {
  inner_name: String!
}

type ProtocGenGraphqlTestBasic_Empty {
  _empty: Boolean
}

enum ProtocGenGraphqlTestBasic_EnumType {
  UNKNOWN
  FIRST
  SECOND
}

enum ProtocGenGraphqlTestBasic_UnusedEnum {
  ZERO
}
